velox/type/Filter.cpp

```diff
@@ -918,6 +920,56 @@ bool BigintValuesUsingHashTable::testInt64Range(
   return max >= *it;
 }
 
+folly::dynamic HugeintValuesUsingHashTable::serialize() const {
+  auto obj = Filter::serializeBase("HugeintValuesUsingHashTable");
+  obj["min_lower"] = HugeInt::lower(min_);
+  obj["min_upper"] = HugeInt::upper(min_);
+  obj["max_lower"] = HugeInt::lower(max_);
+  obj["max_upper"] = HugeInt::upper(max_);
+
+  folly::dynamic lowerValues = folly::dynamic::array;
+  folly::dynamic upperValues = folly::dynamic::array;
+  for (auto v : values_) {
+    lowerValues.push_back(HugeInt::lower(v));
+    upperValues.push_back(HugeInt::upper(v));
+  }
+  obj["lower_values"] = lowerValues;
+  obj["upper_values"] = upperValues;
+
+  return obj;
+}
+
+FilterPtr HugeintValuesUsingHashTable::create(const folly::dynamic& obj) {
+  auto nullAllowed = deserializeNullAllowed(obj);
+  auto min = HugeInt::build(obj["min_upper"].asInt(), obj["min_lower"].asInt());
+  auto max = HugeInt::build(obj["max_upper"].asInt(), obj["max_lower"].asInt());
+  auto values = deserializeHugeintValues(obj);
+
+  return std::make_unique<HugeintValuesUsingHashTable>(
+      min, max, values, nullAllowed);
+}
+
+bool HugeintValuesUsingHashTable::testingEquals(const Filter& other) const {
```

Comment:
Can we template and re-use for BigInt and HugeInt?
Probably use `std::is_same_v` for the cast?

Reply:
Updated, thanks!